x/group/simulation/operations.go

```diff
@@ -78,6 +79,26 @@ const (
 	WeightMsgCreateGroupWithPolicy           = 50
 )
 
+// SharedState shared state between message invocations
+type SharedState struct {
+	minGroupID atomic.Uint64
+}
+
+// NewSharedState constructor
+func NewSharedState() *SharedState {
```

Comment:
Why do you think if we create an interface for this state, like that, we preserve the encapsulation of the struct and do not expose it to possible state changes in the code that may be error-prone?